artemis/src/main/java/tech/pegasys/artemis/DepositCommand.java

```diff
@@ -81,44 +99,144 @@ public void generate(
       STDOUT.log(
           Level.FATAL,
           "Failed to send deposit transaction: " + t.getClass() + ": " + t.getMessage());
-      System.exit(1); // Web3J creates a non-daemon thread we can't shut down. :(
+      exit(1);
     }
-    System.exit(0); // Web3J creates a non-daemon thread we can't shut down. :(
+    exit(0);
+  }
+
+  @VisibleForTesting
+  void waitForTransactionReceipts(final List<SafeFuture<TransactionReceipt>> futures)
+      throws InterruptedException, ExecutionException, TimeoutException {
+    SafeFuture.allOf(futures.toArray(SafeFuture[]::new)).get(2, TimeUnit.MINUTES);
+  }
+
+  @VisibleForTesting
+  void exit(final int status) {
+    System.exit(status); // Web3J creates a non-daemon thread we can't shut down. :(
+  }
+
+  private String getYamlFormattedString(final Pair<BLSKeyPair, BLSKeyPair> keyPair) {
+    return String.format(
+        "- {privkey: '%s', pubkey: '%s', withdrawalPrivkey: '%s', withdrawalPubkey: '%s'}",
+        keyPair.getLeft().getSecretKey().getSecretKey().toBytes(),
+        keyPair.getLeft().getPublicKey().toBytesCompressed(),
+        keyPair.getRight().getSecretKey().getSecretKey().toBytes(),
+        keyPair.getRight().getPublicKey().toBytesCompressed());
+  }
+
+  private void generateKeystores(
+      final Pair<BLSKeyPair, BLSKeyPair> keyPair,
+      final Path keyStoreOutputDirectory,
+      final int counter) {
+    generateKeystore(keyPair.getLeft(), keyStoreOutputDirectory, "validator", counter);
+    generateKeystore(keyPair.getRight(), keyStoreOutputDirectory, "withdrawal", counter);
+  }
+
+  private void generateKeystore(
+      final BLSKeyPair key, final Path outputDir, final String prefix, final int counter) {
+    try {
+      final KdfParam kdfParam = new SCryptParam(32, Bytes32.random());
+      final Cipher cipher = new Cipher(CipherFunction.AES_128_CTR, Bytes.random(16));
+      final String randomPassword = generateRandomHexToken();
+      final KeyStoreData keyStoreData =
+          KeyStore.encrypt(
+              key.getSecretKey().getSecretKey().toBytes(),
+              generateRandomHexToken(),
+              "",
+              kdfParam,
+              cipher);
+
+      // create sub directory
+      final Path keystoreDirectory =
+          Files.createDirectories(outputDir.resolve("validator_" + counter));
+
+      // save keystore
+      KeyStoreLoader.saveToFile(
+          keystoreDirectory.resolve(prefix + "_keystore_" + counter + ".json"), keyStoreData);
+
+      // save password
+      Files.writeString(
+          keystoreDirectory.resolve(prefix + "_password_" + counter + ".txt"), randomPassword);
+    } catch (final IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  private String generateRandomHexToken() {
+    final SecureRandom secureRandom = SecureRandomProvider.createSecureRandom();
+    byte[] token = new byte[32];
+    secureRandom.nextBytes(token);
+    return new BigInteger(1, token).toString(16); // hex encoding
```

Comment:
It would be much better to use:
```suggestion
    return Bytes.wrap(token).toHexString()
```

Reply:
This method has been removed.